### ISSUE TITLE
Directory placement difference in name patterns

### ISSUE THREAD

**Issue by boris-kolpackov** [build2/build2#453]
In our name format it doesn't matter where the directory is placed:

```
file{foo/bar.txt}
foo/file{bar.txt}
```

However, in patters it appears to make a difference, as reported here: https://cpplang.slack.com/archives/CDJ0Z991S/p1734364184322919

```
for f: file{$src_root/dist/taskforce/**.*...}
```

```
buildfile:19:22: error: invalid path '/var/home/ryan/projects/com/TaskForce/dist/taskforce/subdir' in name pattern.
```

Would be good to understand why and hopefully fix.

**Comment by helmesjo** [comment]
Just to clarify, it (should) _not_ have any difference in behavior? I tend to always keep it on the lhs (and avoid pointing to different directories within the same target/prerequisite `{}`).

**Comment by boris-kolpackov** [comment]
> Just to clarify, it (should) not have any difference in behavior (that is, ever)?

Yes, ideally, but as this report shows, there are corner cases (I am sure there are others). Generally, the LHS format is considered "canonical" since it allows you to specify several target types in the same directory, add subdirectories, etc. Consider:

```
foo/{hxx cxx}{* subdir/*}
```

**Comment by boris-kolpackov** [comment]
I believe this is fixed so closing the issue.
